llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp

```diff
@@ -3578,15 +3578,22 @@ SDValue AMDGPUTargetLowering::LowerFP_TO_FP16(SDValue Op, SelectionDAG &DAG) con
     return SDValue();
   }
 
-  assert(N0.getSimpleValueType() == MVT::f64);
+  return LowerF64ToF16(N0, Op.getValueType(), DL, DAG);
```

Comment:
I think `Op.getValueType()` will always be `i32`, right? So you could change `LowerF64ToF16` to always build an `i32` result, and just truncate it to `i16` in `lowerFP_ROUND`.

Reply:
updated